### ISSUE TITLE
Don't overwrite user aliases upon installation

### ISSUE THREAD

**Issue by arxanas** [arxanas/git-branchless#219]
See #218. Although we no longer overwrite aliases, apparently our config file takes precedence over the user one, so we still end up overwriting them in practice. (But at least the aliases can be safely removed later when uninstalling.)

One further complication is that we may want to update git-branchless aliases which already exist upon a fresh run of `init`, just not user aliases.

**Comment by typesanitizer** [comment]
I was thinking the way to fix this would be to include other configuration files after including `branchless/config`.

```
[include]
        path = branchless/config
        path = "$PREFIX/etc/gitconfig" 
        path = "$XDG_CONFIG_HOME/git/config"
        path = "$HOME/.gitconfig"
```

(I'm not sure whether the above will work for Windows though; I was just following the `git-config` docs.)

Does that sound like a reasonable fix? Not entirely sure what complication you're referring to in:

> One further complication is that we may want to update git-branchless aliases which already exist upon a fresh run of init, just not user aliases.

IIUC this should work without problems if you add/delete aliases in `.git/branchless/config`, without having to care about which aliases are user defined, so long as you've added paths as above.

**Comment by arxanas** [comment]
That would be ideal, but I don't know if

* libgit2 (or even the Git CLI) lets you control where to put the include
* if multiple non-consecutive include sections are allowed, which would potentially complicate the semantics

Actually, I suppose the include should go at the top of the file by default, so that any repo-specific aliases are also respected.

**Comment by typesanitizer** [comment]
Yeah, I looked at it a bit more closely, but I don't think doing this using the API or CLI is possible.

We could just modify the file "by hand". Maybe we include comments as section markers.
```
# git-branchless section start
[include]
  # includes added by git-branchless
# git-branchless section end
# old configuration follows...
```
The section markers would make it easier to make the initial setup idempotent (which I understand is the case today).

It feels a little hacky, but I don't have any other bright ideas. Happy to submit a patch with an implementation if you're agree that's the right call.

**Comment by arxanas** [comment]
@typesanitizer Will it still work if the alias is defined in the user's global `.gitconfig`, but the includes are defined in a repo-specific one? My understanding was that the git-branchless aliases would still override the global aliases, regardless of where they're placed in the file.

**Comment by typesanitizer** [comment]
Yep, it should still work; I can include a test case or two 😄. In my testing, included configuration files get the priority of the file they were included in. So those aliases would gain higher priority.

**Comment by typesanitizer** [comment]
I think the current behavior kinda' makes sense intuitively. If I included `~/foo` in the middle of `repo/.git/config`, I'd expect it to override any preceding settings in `~/foo` and be overriden by any later settings. It just so happens in this case that we're picking `foo` = `.gitconfig`, but it would be a bit odd if the behavior changed because of that.

**Comment by arxanas** [comment]
Implemented in #267, thanks @rslabbert!
